File: t2-3/stack.cpp
#include <iostream>
#include "stack.h"
#include "list.h"

struct Stack
{
    List *list;
};

Stack *createStack()
{
    Stack *stack = new Stack;
    stack->list = createList();
    return stack;
}

int sizeStack(Stack *stack)
{
    return sizeList(stack->list);
}

void push(Stack *stack, int value)
{
    addElementTo(stack->list, value, 0);
}

int pop(Stack *stack)
{
    int result = peekListElement(stack->list, 0);
    deleteElementFrom(stack->list, 0);
    return result;
}

void printStack(Stack *stack)
{
    printList(stack->list);
}

bool isEmpty(Stack *stack)
{
    return(sizeStack(stack) == 0);
}

File: sem2/p1-1/src/com/group144/kuzmin/Stack.java
package com.group144.kuzmin;

public class Stack {
    private List list = new List();

    public void push(int value) {
        int size = list.size();
        list.add(value, size);
    }

    public int peek() {
        if (list.isEmpty())
            return -1;

        return list.peek(0);
    }

    public int pop() {
        if (list.isEmpty())
            return -1;

        int result = list.peek(0);
        list.delete(0);
        return result;
    }

    public void print() {
        list.print();
    }

    public boolean isEmpty() {
        return list.isEmpty();
    }

    public int size() {
        return list.size();
    }
}
